app/components/post_list/post/footer/footer.tsx

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {TouchableOpacity, View} from 'react-native';
+
+import {updateThreadFollowing} from '@actions/remote/thread';
+import CompassIcon from '@components/compass_icon';
+import FormattedText from '@components/formatted_text';
+import UserAvatarsStack from '@components/user_avatars_stack';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+import type UserModel from '@typings/database/models/servers/user';
+
+type Props = {
+    currentUserId: string;
+    participants: UserModel[];
+    serverUrl: string;
+    teamId: string;
+    teammateNameDisplay: string;
+    testID: string;
+    theme: Theme;
+    thread: ThreadModel;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    const followingButtonContainerBase = {
+        justifyContent: 'center',
+        height: 32,
+        paddingHorizontal: 12,
+    };
+
+    return {
+        container: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            minHeight: 40,
+        },
+        avatarsContainer: {
+            marginRight: 12,
+            paddingVertical: 8,
+        },
+        replyIconContainer: {
+            top: -1,
+            marginRight: 5,
+        },
+        replies: {
+            alignSelf: 'center',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            marginRight: 12,
+            ...typography('Heading', 75),
+        },
+        notFollowingButtonContainer: {
+            ...followingButtonContainerBase,
+            paddingLeft: 0,
+        },
+        notFollowing: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Heading', 75),
+        },
+        followingButtonContainer: {
+            ...followingButtonContainerBase,
+            backgroundColor: changeOpacity(theme.buttonBg, 0.08),
+            borderRadius: 4,
+        },
+        following: {
+            color: theme.buttonBg,
+            ...typography('Heading', 75),
+        },
+        followSeparator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.16),
+            height: 16,
+            marginRight: 12,
+            width: 1,
+        },
+    };
+});
+
+const Footer = ({currentUserId, participants, serverUrl, teamId, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const styles = getStyleSheet(theme);
+    const onUnfollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, false);
+    }), []);
+
+    const onFollow = useCallback(preventDoubleTap(() => {
+        updateThreadFollowing(serverUrl, teamId, thread.id, true);
+    }), []);
```

Comment:
The two functions can be combined.  `updateThreadFollowing` can read from DB, extract the current value for `state` and then send the opposite to the server and local.

Reply:
Or maybe, just use `!thread.isFollowing` instead of true/false